aws/aws.go

```diff
@@ -53,19 +56,33 @@ func NewHandler(config *Config) (*Handler, error) {
 
 func (awsHandler *Handler) initAws() error {
 	log.Println("Creating AWS session...")
-	var err error
-	creds := credentials.NewStaticCredentials(awsHandler.config.AwsAccessKeyID, awsHandler.config.AwsSecretKey, "")
 
-	awsHandler.awsSession, err = session.NewSession(&aws.Config{
-		Region:      aws.String(awsHandler.config.AwsRegion),
-		Credentials: creds,
-	})
-	if err != nil {
-		return err
+	// to prevent a breaking change of supplying the AWS Creds/Region via
+	// arguments, but not in the env, push them back to the env to be picked
+	// up by the session
+	if len(awsHandler.config.AwsAccessKeyID) > 0 {
+		os.Setenv("AWS_ACCESS_KEY_ID", awsHandler.config.AwsAccessKeyID)
+	}
+	if len(awsHandler.config.AwsSecretKey) > 0 {
+		os.Setenv("AWS_SECRET_KEY", awsHandler.config.AwsSecretKey)
+	}
+	if len(awsHandler.config.AwsRegion) > 0 {
+		os.Setenv("AWS_REGION", awsHandler.config.AwsRegion)
 	}
```

Comment:
Let's fire off some deprecation warnings if any of these are set. It's quite insecure to use command line flags for secrets, so we should hint that a better way is possible.

Let's aim to rid ourselves of these flags entirely, eventually. Ideally, a 1.0 release should not include them.

Reply:
@echlebek I've added deprecation warnings via log message, help output, and the README.